pkg/pod/script.go

```diff
@@ -132,7 +162,85 @@ cat > ${tmpfile} << '%s'
 		// we'll clear out the Args and overwrite Command.
 		steps[i].Command = []string{tmpFile}
 		steps[i].VolumeMounts = append(steps[i].VolumeMounts, scriptsVolumeMount)
+
+		// Add debug mounts if breakpoints are present
+		if len(breakpoints) > 0 {
+			debugInfoVolumeMount := corev1.VolumeMount{
+				Name:      debugInfoVolumeName,
+				MountPath: filepath.Join(debugInfoDir, fmt.Sprintf("%d", i)),
+			}
+			steps[i].VolumeMounts = append(steps[i].VolumeMounts, debugScriptsVolumeMount, debugInfoVolumeMount)
+		}
 		containers = append(containers, steps[i].Container)
 	}
+
+	// Place debug scripts if breakpoints are enabled
+	if len(breakpoints) > 0 {
+		// Debug script names
+		debugContinueScriptName := "continue"
+		debugFailContinueScriptName := "fail-continue"
+
+		// debugContinueScript can be used by the user to mark the failing step as a success
+		debugContinueScript := defaultScriptPreamble + fmt.Sprintf(`
+numberOfSteps=%d
+debugInfo=%s
+tektonTools=%s
+
+postFile="$(ls ${debugInfo} | grep -E '[0-9]+' | tail -1)"
+stepNumber="$(echo ${postFile} | sed 's/[^0-9]*//g')"
+
+if [ $stepNumber -lt $numberOfSteps ]; then
+	touch ${tektonTools}/${stepNumber} # Mark step as success
+	echo "0" > ${tektonTools}/${stepNumber}.breakpointexit
+	echo "Executing step $stepNumber..."
+else
+	echo "Last step (no. $stepNumber) has already been executed, breakpoint exiting !"
+	exit 0
+fi`, len(steps), debugInfoDir, mountPoint)
+
+		// debugContinueScript can be used by the user to mark the failing step as a success
+		debugFailContinueScript := defaultScriptPreamble + fmt.Sprintf(`
```

Comment:
This big block of bash should be a package-level `const` at least, and possibly a separate file that's `//go:embed`ed. It might also benefit from using text/template instead of positional `%s` placeholders.

Reply:
I have added them as package level constants but I will open an issue to move this to a template usage. Created https://github.com/tektoncd/pipeline/issues/3974 to track this further.